connector/src/yang/connector/grpc/telegraf.py

```diff
@@ -0,0 +1,159 @@
+                    with open(self.config_file, 'w') as f:
+                        log.info(f"Creating telegraf config file {self.config_file}")
+                        f.write('')
+
+                    # create default config
+                    # global tags
+                    config.add_section('global_tags')
+                    config.set('global_tags', 'user', r'"${USER}"')
+
+                    # input configuration
+                    config.add_section('[inputs.cisco_telemetry_mdt]')
+                    config.set('[inputs.cisco_telemetry_mdt]', 'transport', '"grpc"')
+                    config.set('[inputs.cisco_telemetry_mdt]', 'service_address', f'":{allocated_port}"')
+
+                    # default output config - to file in runtime or user supplied dir
+                    config.add_section('[outputs.file]')
+                    config.set('[outputs.file]', 'files', f'["stdout", "{self.output_file}"]')
+                    config.set('[outputs.file]', 'data_format', '"json"')
+                    config.set('[outputs.file]', 'json_timestamp_units', '"1ms"')
+                    config.set('[outputs.file]', 'rotation_max_size', '"2048MB"')
+                    config.set('[outputs.file]', 'flush_jitter', '"500ms"')
+
+                    # apply config
+                    with open(self.config_file, 'w') as f:
+                        log.info(f"Updating {self.config_file}")
+                        config.write(f)
+
+        # exit context manager to release port
+        # spawn telegraf/pipeline using config
+        if subprocess.run(['which', 'telegraf']).returncode == 0:
+            self.transport_process = subprocess.Popen(f"telegraf -config '{self.config_file}'", shell=True)
+        else:
+            raise OSError('Telegraf is not installed')
+        # log port
+        log.info(f"Telegraf is running as PID {self.transport_process.pid} on port {allocated_port}")
+
+        # check if there is an existing unicon connection
+        active_connection = None
+        for conn_alias in self.device.connectionmgr.connections:
+            conn = self.device.connectionmgr.connections[conn_alias]
+            if isinstance(conn, Connection) and conn.connected:
+                active_connection = conn_alias
+                break
+        # create one if there isn't
+        if not active_connection:
+            self.device.instantiate()
+            if 'unicon' in self.device.default.__module__:
+                self.device.connect()
+            else:
+                raise ValueError('Connection Class is not Unicon')
+            active_connection = self.device.connectionmgr.connections._default_alias
+
+        # run configurations while ensuring that it is using a unicon default connection
+        with self.device.temp_default_alias(active_connection):
+            local_ip = self.transporter_ip or self.device.api.get_local_ip()
+            if self.telemetry_autoconfigure:
```

Comment:
Please move this check to line 120

Reply:
Moving this check will break the logging as autoconfigure being false will not generate a local_ip from the api